src/Exception/CircularWeavingException.php

```diff
@@ -0,0 +1,16 @@
+<?php
+/*
+ * Go! AOP framework
+ *
+ * @copyright Copyright 2014, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\Exception;
+
+class CircularWeavingException extends \RuntimeException implements ExceptionInterface
```

Comment:
I was playing around with https://github.com/goaop/goaop-symfony-bundle/issues/15, because I can easily configure new environment for functional tests, this was easy pick now - I have created new exception class (btw - there are no custom exceptions in library, maybe we should add?)